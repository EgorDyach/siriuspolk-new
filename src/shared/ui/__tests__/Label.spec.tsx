import { render, screen } from '@testing-library/react';
import { Label } from '../Label';

describe('<Label />', () => {
  it('should render correctly', () => {
    render(<Label>test label</Label>);
    expect(screen.getByText(/test label/i)).toBeInTheDocument();
  });

  it('should render correctly with classname', () => {
    render(<Label className="test-class" />);
    expect(screen.getByTestId('label')).toHaveClass('test-class');
  });

  it('should render correctly with ReactNode', () => {
    render(
      <Label>
        <h1>heading</h1>
        <button>click</button>
      </Label>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
  it('matches snapshot with default label', () => {
    const { asFragment } = render(<Label>test label</Label>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with react node', () => {
    const { asFragment } = render(
      <Label>
        <h1>heading</h1>
        <button>click</button>
      </Label>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
