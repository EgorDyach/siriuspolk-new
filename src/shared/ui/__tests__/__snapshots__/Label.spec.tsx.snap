// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> matches snapshot with default label 1`] = `
<DocumentFragment>
  <label
    class="flex items-center gap-1 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
    data-slot="label"
    data-testid="label"
  >
    test label
  </label>
</DocumentFragment>
`;

exports[`<Label /> matches snapshot with react node 1`] = `
<DocumentFragment>
  <label
    class="flex items-center gap-1 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
    data-slot="label"
    data-testid="label"
  >
    <h1>
      heading
    </h1>
    <button>
      click
    </button>
  </label>
</DocumentFragment>
`;
